students/drovdahl/lesson03/slicing.py

```diff
@@ -0,0 +1,148 @@
+#! /usr/local/bin/python3
+
+
+'''
+Write some functions that take a sequence as an argument and return a copy of
+that sequence with modifications:
+
+ with the first and last items exchanged.
+ with every other item removed.
+ with the first and last 4 items removed, and every other item in between.
+ with the elements reversed (just with slicing).
+ with the last third, then first third, then the middle third in the new order.
+'''
+
+a_string = 'this is a really really long string'
+a_tuple = (2, 54, 13, 12, 5, 32, 18, 6, 9, 1, 88, 60)
+a_list = [2, 54, 13, 12, 5, 32, 18, 6, 9, 1, 88, 60]
+
+
+def first_last_flip(seq):
+    '''
+    This function exchanges the first and last elements of the sequence arg
+    It works with tuples, lists, and string sequences
+    '''
+    # process for tuples
+    while type(seq) is tuple:
```

Comment:
Slicing always returns a copy of the slice, so I don't think we need to distinguish between tuple/lists. You just need line 39-40, nothing else in this function. Same w/ other functions below.